google/cloud/ndb/tasklets.py

```diff
@@ -25,17 +25,35 @@
 the tasklet, any yield of a Future waits for and returns the Future's result.
 For example::
 
+    from tasklets import tasklet, Return
+
     @tasklet
     def foo():
-        a = yield <some Future>
-        b = yield <another Future>
-        return a + b
+        a = yield <AFuture>
+        b = yield <BFuture>
+        raise Return(a + b)
 
     def main():
         f = foo()
         x = f.result()
         print x
 
+In this example, `foo` needs the results of two futures, `AFuture` and
+`BFuture`, which it gets somehow, for example as results of calls.
+Rather than waiting for their values and blocking, it yields. First,
+the tasklet yields `AFuture`.  The event loop gets `AFuture` and takes
+care of waiting for its result.  When the event loop gets the result
+of `AFuture`, it sends it to the tasklet by calling `send` on the
+iterator returned by calling the tasklet.  The tasklet assigns the
+value sent to `a` and then yields `BFuture`.  Again the event loop
+waits for the result of `BFuture` and sends it to the tasklet.  The
+tasklet then has what it needs to compute a result.
+
+The tasklet returns its result by raising a `Result` exception.  This
+is needed because values returned by generators aren't available to
+callers in Python 2.  When support for Python 2 is dropped, then
+tasklets will be able to just return their results.
+
```

Comment:
We are forced to use `raise Return(foo)` internally, because *we* have to support to Python 2, but our users are not similarly constrained. If they are using Python 3 and lay no claim to supporting Python 2, they are free now to use `return foo`.

Reply:
Do users write tasklets?

In any case, I'll add that for applications using Python 3, they should just return values.

Reply:
Yes, tasklets are intended to be a user facing feature so that users can write async code.

https://cloud.google.com/appengine/docs/standard/python/ndb/async

Reply:
Specifically, this doctstring is part of the public, user facing documentation:

https://googleapis.dev/python/python-ndb/latest/tasklets.html

Reply:
Do we know that users are writing their own tasklets? :)

Reply:
In any case, I reworded based on your comments. See what you think.